neo/SmartContract/InteropService.cs

```diff
@@ -143,8 +143,36 @@ private static bool Runtime_GetTrigger(ApplicationEngine engine)
 
         internal static bool CheckWitness(ApplicationEngine engine, UInt160 hash)
         {
-            var _hashes_for_verifying = engine.ScriptContainer.GetScriptHashesForVerifying(engine.Snapshot);
-            return _hashes_for_verifying.Contains(hash);
+            if (engine.ScriptContainer is Transaction tx)
+            {
+                Cosigner usage = tx.Cosigners.FirstOrDefault(p => p.Account.Equals(hash));
+                if (usage is null) return false;
+                if (usage.Scopes.HasFlag(WitnessScope.Global))
+                    return true;
+                if (usage.Scopes.HasFlag(WitnessScope.CalledByEntry))
+                {
+                    if (engine.CallingScriptHash == engine.EntryScriptHash)
+                        return true;
```

Comment:
If we treated it as a flag, but the first find we returned `true`, maybe we should remove the flag

Reply:
Why `AND`?

Reply:
If I fill `AllowedContracts` and `AllowedGroups` , i expect that both of them was checked,

Reply:
I don't think so. If you set `AllowedContracts`, you know the contract's `ScriptHash`, so why would you need to specify a group?

Reply:
Agree, but then we should remove the flag, because have no sense the combination

Reply:
@erikzhang I also thought it was an AND... if user specify custom hash and group, it would expect both to be fullfilled simultaneously. However.... in the current situation, we cannot specify an OR due to the lack of repetition on Account, and OR is perhaps more important than AND, I dont know.
We can specify an OR between hashes (because its a multi field), and also between groups. If its definitely an OR, we will be able to allow Entry and groups simultaneously, what may be useful, same thing for Entry OR specific hashes. Global + OR is useless, but Global + AND is also useless :joy:
Due to the simplicity of the design (by unique Accounts, which is good), I will vote for OR, and not support AND. I think this feature is already powerful enough for people designing their security, the manifest is a native AND for all extra things.

Reply:
@shargon Entry + OR may useful... consider an asset paid game, where you need the Entry (to protect your native assets paid), and the OR to "unlock" your own assets during the game.
OR is much more important than AND.

Reply:
I added some tests to reinforce this case @shargon .